jobs/cloud-provider/templates/bin/cloud-provider_utils.erb

```diff
@@ -1,14 +1,5 @@
 #!/bin/bash -ex
 
-set_node_name() {
-  node_name="<%= spec.ip %>"
-<% if_p("cloud-provider.type") do | provider_type | %>
-  <% if provider_type == "gce" %>
-  node_name="$(curl http://metadata.google.internal/computeMetadata/v1/instance/name -H "Metadata-Flavor: Google").c.<%= p("cloud-provider.gce.project-id") %>.internal"
-  <% end %>
-<% end %>
-}
```

Comment:
This is necessary on GCP, the node name needs to follow the format vm-name.c.project-id.internal. This is because it needs to be a resolvable address and it needs to begin with the name of the vm. Otherwise Kubernetes won't be able to find the VM in order to attach a disk to it.